tests/manage/pv_services/test_resource_deletion_during_pvc_pod_deletion_and_io.py

```diff
@@ -0,0 +1,335 @@
+
+        # Start IO on pods to be deleted
+        log.info("Starting IO on pods to be deleted.")
+        for pod_obj in pods_to_delete:
+            pod_obj.run_io(storage_type='fs', size=f'{self.pvc_size - 1}G')
+        log.info("IO started on pods to be deleted.")
+
+        # Start deleting PVCs
+        pvc_bulk_delete = executor.submit(delete_pvcs, pvcs_to_delete)
+        log.info("Started deleting PVCs")
+
+        # Start deleting pods
+        pod_bulk_delete = executor.submit(self.delete_pods, pods_to_delete)
+        log.info("Started deleting pods")
+
+        # Start IO on IO pods
+        for pod_obj in io_pods:
+            pod_obj.run_io(storage_type='fs', size=f'{self.pvc_size - 1}G')
+        log.info("Started IO on IO pods")
+
+        # Verify pvc deletion has started
+        pvc_deleting = executor.submit(
+            wait_for_resource_count_change, func_to_use=get_all_pvcs,
+            previous_num=initial_num_of_pvc, namespace=namespace,
+            change_type='decrease', min_difference=1, timeout=30, interval=0.01
+        )
+
+        # Verify pod deletion has started
+        pod_deleting = executor.submit(
+            wait_for_resource_count_change, func_to_use=get_all_pods,
+            previous_num=initial_num_of_pods, namespace=namespace,
+            change_type='decrease', min_difference=1, timeout=30, interval=0.01
+        )
+
+        assert pvc_deleting.result(), (
+            "Wait timeout: PVCs are not being deleted."
+        )
+        logging.info(
+            f"PVCs deletion has started."
+        )
+
+        assert pod_deleting.result(), (
+            "Wait timeout: Pods are not being deleted."
+        )
+        logging.info(
+            f"Pods deletion has started."
+        )
+
+        # Delete pod of type 'resource_to_delete'
+        disruption.delete_resource()
+
+        pods_deleted = pod_bulk_delete.result()
+        assert pods_deleted, "Deletion of pods failed."
+
+        # Verify pods are deleted
+        for pod_obj in pods_to_delete:
+            assert pod_obj.ocp.wait_for_delete(pod_obj.name), (
+                f"Pod {pod_obj.name} is not deleted"
+            )
+        logging.info("Verified: Pods are deleted.")
+
+        # Verify that the mount point is removed from nodes after deleting pod
+        node_pv_mounted = verify_pv_mounted_on_node(node_pv_dict)
+        for node, pvs in node_pv_mounted.items():
+            assert not pvs, (
+                f"PVs {pvs} is still present on node {node} after "
+                f"deleting the pods."
+            )
+        log.info(
+            "Verified: mount points are removed from nodes after deleting "
+            "the pods"
+        )
+
+        pvcs_deleted = pvc_bulk_delete.result()
+        assert pvcs_deleted, "Deletion of PVCs failed."
+
+        # Verify PVCs are deleted
+        for pvc_obj in pvcs_to_delete:
+            assert pvc_obj.ocp.wait_for_delete(pvc_obj.name), (
```

Comment:
same

Reply:
It will return bool only on success.
Removed assert